sdk_v2/kubeflow/training/api/training_client.py

```diff
@@ -12,7 +12,388 @@
+        train_func: Optional[Callable] = None,
+        num_nodes: Optional[int] = None,
+        resources_per_node: Optional[dict] = None,
+        packages_to_install: Optional[List[str]] = None,
+        pip_index_url: str = constants.DEFAULT_PIP_INDEX_URL,
+        # TODO (andreyvelich): Add num_nodes, func, resources to the Trainer or TrainerConfig ?
+        trainer_config: Optional[types.TrainerConfig] = None,
+        dataset_config: Optional[types.HuggingFaceDatasetConfig] = None,
+        model_config: Optional[types.HuggingFaceModelInputConfig] = None,
+    ) -> str:
+        """Create the TrainJob. TODO (andreyvelich): Add description
+
+        Returns:
+            str: The unique name of the TrainJob that has been generated.
+
+        Raises:
+            ValueError: Input arguments are invalid.
+            TimeoutError: Timeout to create TrainJobs.
+            RuntimeError: Failed to create TrainJobs.
+        """
+
+        # Generate unique name for the TrainJob.
+        # TODO (andreyvelich): Discuss this TrainJob name generation.
+        train_job_name = random.choice(string.ascii_lowercase) + uuid.uuid4().hex[:11]
+
+        # Build the Trainer.
+        trainer = models.KubeflowOrgV2alpha1Trainer()
+
+        # Add number of nodes to the Trainer.
+        if num_nodes is not None:
+            trainer.num_nodes = num_nodes
+
+        # Add resources per node to the Trainer.
+        if resources_per_node is not None:
+            trainer.resources_per_node = utils.get_resources_per_node(
+                resources_per_node
+            )
+
+        # Add command and args to the Trainer if training function is set.
+        if train_func is not None:
+            trainer.command = constants.DEFAULT_COMMAND
+            # TODO: Support train function parameters.
+            trainer.args = utils.get_args_using_train_func(
+                train_func,
+                None,
+                packages_to_install,
+                pip_index_url,
+            )
+
+        # Add the Lora config to the Trainer envs.
+        if trainer_config and trainer_config.lora_config:
+            trainer.env = utils.get_lora_config(trainer_config.lora_config)
+
+        train_job = models.KubeflowOrgV2alpha1TrainJob(
+            api_version=constants.API_VERSION,
+            kind=constants.TRAINJOB_KIND,
+            metadata=client.V1ObjectMeta(name=train_job_name),
+            spec=models.KubeflowOrgV2alpha1TrainJobSpec(
+                runtime_ref=models.KubeflowOrgV2alpha1RuntimeRef(name=runtime_ref),
+                trainer=(
+                    trainer if trainer != models.KubeflowOrgV2alpha1Trainer() else None
+                ),
+                dataset_config=utils.get_dataset_config(dataset_config),
+                model_config=utils.get_model_config(model_config),
+            ),
+        )
+
+        # Create the TrainJob.
+        try:
+            self.custom_api.create_namespaced_custom_object(
```

Comment:
should we check the status of this api, whether this returns 200 or not

Reply:
I think, if this API fails it throws an Exception
https://github.com/kubeflow/training-operator/blob/e9ff7d1a93822092595be7d0149685ea0eb3cbfe/sdk_v2/kubeflow/training/api/training_client.py#L205-L209